zb-db/src/main/java/io/zeebe/db/impl/rocksdb/RocksDbConfiguration.java

```diff
@@ -12,36 +12,56 @@
 public final class RocksDbConfiguration {
 
   public static final long DEFAULT_MEMORY_LIMIT = 512 * 1024 * 1024L;
+  public static final int DEFAULT_UNLIMITED_MAX_OPEN_FILES = -1;
 
   private final Properties columnFamilyOptions;
   private final boolean statisticsEnabled;
   private final long memoryLimit;
 
+  /**
+   * Defines how many files are kept open by RocksDB, per default it is unlimited (-1). This is done
+   * for performance reasons, if we set a value higher then zero it needs to keep track of open
+   * files in the TableCache and look up on accessing them.
+   *
+   * <p>https://github.com/facebook/rocksdb/wiki/RocksDB-Tuning-Guide#general-options
+   */
+  private final int maxOpenFiles;
+
   private RocksDbConfiguration(
       final Properties columnFamilyOptions,
       final boolean statisticsEnabled,
-      final long memoryLimit) {
+      final long memoryLimit,
+      final int maxOpenFiles) {
     this.columnFamilyOptions = columnFamilyOptions;
     this.statisticsEnabled = statisticsEnabled;
     this.memoryLimit = memoryLimit;
+    this.maxOpenFiles = maxOpenFiles;
   }
 
   public static RocksDbConfiguration empty() {
-    return new RocksDbConfiguration(new Properties(), false, DEFAULT_MEMORY_LIMIT);
+    return of(new Properties());
   }
 
   public static RocksDbConfiguration of(final Properties properties) {
-    return new RocksDbConfiguration(properties, false, DEFAULT_MEMORY_LIMIT);
+    return of(properties, false);
   }
 
   public static RocksDbConfiguration of(
```

Comment:
Nit: this seems only used by the previous method - I would just remove it for now since who knows what this will look like in the future.

Reply:
No this is also used in the engine :)

Reply:
Yeah this chaining makes it a bit easier to maintain it, otherwise I always have to change the values everywhere. If I remove both methods I need for of(Properties) and empty() to call the ctor with the same values.